### PR TITLE
GRW-2098 - feat(AccordionBlock): added support for FAQ structured data

### PR DESCRIPTION
**Pull request by guilhermespopolin**
## Describe your changes

* Added [FAQ structured data](https://developers.google.com/search/docs/appearance/structured-data/faqpage) for `AccordionBlock`. It was based on `market-web`'s [implementation](https://github.com/HedvigInsurance/market-web/blob/master/src/utils/structuredData.ts).

[Add Google Rich Results preview here]

## Justify why they are needed

Requested by Peter for SEO purposes.

## Jira issue(s): [GRW-2098](https://hedvig.atlassian.net/browse/GRW-2098)

[GRW-2098]: https://hedvig.atlassian.net/browse/GRW-2098?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by guilhermespopolin** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1365** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1365" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1365?utm_source=stack-comment).

File: apps/store/src/blocks/AccordionBlock.tsx [hunk 1]
```diff
@@ -62,3 +78,20 @@ const StyledAccordion = styled(Accordion.Root)({
...
+) => {
+  return JSON.stringify({
+    '@context': 'http://schema.org',
+    '@type': 'FAQPage',
+    mainEntity: accordions.map((item) => ({
+      '@type': 'Question',
+      name: item.title,
+      acceptedAnswer: {
+        '@type': 'Answer',
+        text: storyblokClient.richTextResolver.render(item.body),
```
Comment (guilhermespopolin) [inline apps/store/src/blocks/AccordionBlock.tsx:93]: Based on the [docs](https://developers.google.com/search/docs/appearance/structured-data/faqpage#faq-page) for FAQ structured the content Google can display HTML some tags.

File: apps/store/src/blocks/AccordionBlock.tsx [hunk 2]
```diff
@@ -1,42 +1,58 @@
...
+import StoryblokClient from 'storyblok-js-client'
 import { Heading, Text, theme, mq } from 'ui'
 import { AccordionItemBlock, AccordionItemBlockProps } from '@/blocks/AccordionItemBlock'
 import * as Accordion from '@/components/Accordion/Accordion'
 import { ExpectedBlockType, SbBaseBlockProps } from '@/services/storyblok/storyblok'
 import { filterByBlockType } from '@/services/storyblok/Storyblok.helpers'
 
+const storyblokClient = new StoryblokClient({
+  accessToken: process.env.NEXT_PUBLIC_STORYBLOK_ACCESS_TOKEN,
+})
```
Comment (robinandeer) [inline apps/store/src/blocks/AccordionBlock.tsx:13]: Hmm should we really initiate this again here...? Can't we import it from a shared module?

File: apps/store/src/blocks/AccordionBlock.tsx [hunk 3]
```diff
@@ -1,42 +1,58 @@
...
-          )}
-          {blok.description && (
-            <Text color="textSecondary" size={{ _: 'xl', md: 'xxl' }}>
-              {blok.description}
-            </Text>
-          )}
-        </TitleDescriptionWrapper>
+    <>
+      {enableFAQStructuredData && (
+        <Script id="faq-sctructured-data" type="application/ld+json">
```
Comment (robinandeer) [inline apps/store/src/blocks/AccordionBlock.tsx:30]: What if we have multiple FAQ sections on the same page? Then we need unique ID, right? Do we even need to provide ID?

**Review by robinandeer — APPROVED** [review]

File: apps/store/src/blocks/AccordionBlock.tsx (hunk 3 above)
Comment (gustaveen) [inline apps/store/src/blocks/AccordionBlock.tsx:30]: Wondering myself, does it need to be inside `<head>`? Let's validate later 👍

**Review by gustaveen — APPROVED** [review]
Great! 🙌 Just wondering if we need to setup a storyblok-client for rendering the FAQ content

File: apps/store/src/blocks/AccordionBlock.tsx [hunk 4]
```diff
@@ -1,42 +1,58 @@
...
+  items: ExpectedBlockType<AccordionItemBlockProps>
   title?: string
   description?: string
-  items: ExpectedBlockType<AccordionItemBlockProps>
+  isFAQ?: boolean
 }>
 
 export const AccordionBlock = ({ blok }: Props) => {
   const accordionItems = filterByBlockType(blok.items, AccordionItemBlock.blockName)
+  const enableFAQStructuredData = blok.isFAQ ?? false
```
[the remark this answers is not in the document]
Comment (guilhermespopolin) [reply apps/store/src/blocks/AccordionBlock.tsx:24]: It seems the default value for it in `market-web` is false. But I'm gonna double check with him.

File: apps/store/src/blocks/AccordionBlock.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (guilhermespopolin) [reply apps/store/src/blocks/AccordionBlock.tsx:13]: RIght, ofc 🤦🏻

File: apps/store/src/blocks/AccordionBlock.tsx (hunk 3 above)
Comment (guilhermespopolin) [reply apps/store/src/blocks/AccordionBlock.tsx]: Decided to move forward and add it within `<head>`.

File: apps/store/src/blocks/AccordionBlock.tsx (hunk 2 above)
Comment (guilhermespopolin) [reply apps/store/src/blocks/AccordionBlock.tsx]: This will be removed by using `renderRichText` from `@storyblok/react` as Gustav suggested.

**Pull request merged**